### ISSUE TITLE
AttributeManager is not compatible with HDF5 datatypes

### ISSUE THREAD

**Issue by jordan-farq** [h5py/h5py#1476]
The h5py AttributeManager class cannot be used with attributes on an HDF5 datatype. It seems to only be compatible with groups and datasets. An attempt to use the dictionary-style AttributeManager methods on attributes attached to a datatype results in an `AttributeError` with the description that the datatype object `has no attribute 'get_create_plist'`. 

Below is an example of the full stack trace reproducing the error, where the file at `filename` contains a compound datatype named `datatype_name` that has attributes attached to it. As shown, attempting to access the keys produces the error. The same error will occur for `values()`, `items()`, and `__iter__().__next__()`, etc.

```
>>> f = h5py.File(filename, 'r')
>>> datatype = f[datatype_name]
>>> datatype_attrs = datatype.attrs
>>> list(datatype_attrs.keys())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/Caskroom/miniconda/4.6.14/miniconda3/envs/xd-h5py/lib/python3.7/_collections_abc.py", line 720, in __iter__
    yield from self._mapping
  File "/usr/local/Caskroom/miniconda/4.6.14/miniconda3/envs/xd-h5py/lib/python3.7/site-packages/h5py/_hl/attrs.py", line 264, in __iter__
    cpl = self._id.get_create_plist()
AttributeError: 'h5py.h5t.TypeCompoundID' object has no attribute 'get_create_plist'
```
It seems that the `H5Tget_create_plist` function from the HDF5 C library has not been added to the h5py package (the C function in the online HDF5 reference manual is here: https://support.hdfgroup.org/HDF5/doc/RM/RM_H5T.html#Datatype-GetCreatePlist). This functionality could probably be added somewhat easily and would prove useful for many use cases.

System details:
 * Operating System: MacOS Mojave 10.14.5
 * Python version: 3.7.6
 * Where Python was acquired: miniconda
 * h5py version: 2.10.0
 * HDF5 version: 1.10.5

```
>>> print(h5py.version.info)
Summary of the h5py configuration
---------------------------------

h5py    2.10.0
HDF5    1.10.5
Python  3.7.6 | packaged by conda-forge | (default, Jan  7 2020, 22:05:27) 
[Clang 9.0.1 ]
sys.platform    darwin
sys.maxsize     9223372036854775807
numpy   1.17.3
```

**Comment by takluyver** [comment]
Seems reasonable, do you want to make a PR? You'll need to first add a binding in `h5py/api_functions.txt`, and then a method in `h5py/h5t.pyx`. And of course a test that fails beforehand and passes after.

**Comment by jordan-farq** [comment]
@takluyver sure, I can make a PR. Not sure exactly when I can get to it but hopefully over the weekend. The instructions here seem pretty clear: https://github.com/h5py/h5py/blob/master/docs/contributing.rst#how-to-modify-h5py

**Comment by takluyver** [comment]
Fixed by #1491.
